codegen/src/main/kotlin/org/jetbrains/bsp/generators/bsp4rs/SerializationRenderer.kt

```diff
@@ -0,0 +1,129 @@
+package org.jetbrains.bsp.generators.bsp4rs
+
+import org.jetbrains.bsp.generators.dsl.CodeBlock
+import org.jetbrains.bsp.generators.dsl.rustCode
+import org.jetbrains.bsp.generators.ir.*
+
+class SerializationRenderer {
+    private var serdeSet: Set<SerdeOption> = emptySet()
+    private var reprSet: Set<ReprOption> = emptySet()
+
+    fun renderForDef(def: Def, untagged: Boolean = false): CodeBlock {
+        prepareSerdeSet(def.hints)
+        serdeSet = serdeSet.plus(defToSerdeList(def, untagged))
+
+        reprSet = setOf()
+        reprSet = reprSet.plus(defToReprList(def))
+
+        return rustCode {
+            -printSerde()
+            -printRepr()
+        }
+    }
+
+    fun renderForField(field: Field): CodeBlock {
+        prepareSerdeSet(field.hints)
+        serdeSet = serdeSet.plus(fieldToSerdeList(field))
+
+        return rustCode {
+            -printSerde()
+        }
+    }
+
+    private fun prepareSerdeSet(hints: List<Hint>) {
+        val rename = hints.find { it is Hint.JsonRename }
+
+        return if (rename is Hint.JsonRename)
+            serdeSet = setOf(SerdeOption.Rename(rename.name))
+        else
+            serdeSet = emptySet()
+    }
+
+    private fun defToSerdeList(def: Def, untagged: Boolean) = when (def) {
```

Comment:
please add return types everywhere where it's not unit

Reply:
done